domainobjs/ts/__tests__/ballot.test.ts

```diff
@@ -3,22 +3,102 @@ import { expect } from "chai";
 import { Ballot } from "..";
 
 describe("Ballot", () => {
-  it("should create a new ballot and hash it", () => {
-    const b = new Ballot(0, 2);
-    const h = b.hash();
-    expect(h).to.not.eq(null);
+  describe("constructor", () => {
+    it("should create an empty ballot", () => {
+      const b = new Ballot(0, 2);
+      expect(b.votes.length).to.eq(0);
+    });
+
+    it("should create a ballot with 1 vote", () => {
+      const b = new Ballot(1, 2);
+      expect(b.votes.length).to.eq(1);
+      expect(b.votes[0]).to.eq(BigInt(0));
+    });
+  });
+
+  describe("hash", () => {
+    it("should produce an hash of the ballot", () => {
+      const b = new Ballot(0, 2);
+      const h = b.hash();
+      expect(h).to.not.eq(null);
+    });
+  });
+
+  describe("copy", () => {
+    it("should produce a deep copy", () => {
+      const b1 = Ballot.genRandomBallot(2, 2);
+      const b2 = b1.copy();
+      expect(b1.equals(b2)).to.eq(true);
+    });
+  });
```

Comment:
Should we test if the deep copy really works just like what we test in the commands.test.ts L91:104?
Such as 
``` ts
const b1 = Ballot.genRandomBallot(2, 2);

// shallow copy
const b2 = b1;
b1.nonce = BigInt(9999);
expect(b1.nonce.toString()).to.eq(b2.nonce.toString());

// deep copy
const b3 = b1.copy();
b1.nonce = BigInt(8888);
expect(b1.nonce.toString()).not.to.eq(b3.nonce.toString());
```